tiledb/bioimg/converters/base.py

```diff
@@ -316,24 +377,55 @@ def _convert_level_to_tiledb(
     attr_dtype = reader.level_dtype(level)
     schema = get_schema(dim_names, dim_shape, max_tiles, attr_dtype, compressor)
 
+    channel_index = source_axes.dims.find("C")
+    channel_count = source_shape[channel_index] if channel_index > -1 else 1
+    channel_min_max = [
+        [
+            np.iinfo(reader.level_dtype(level)).max,
+            np.iinfo(reader.level_dtype(level)).min,
+        ]
+        for _ in range(channel_count)
+    ]
+
     # get or create TileDB array uri
     uri, created = rw_group.get_or_create(f"l_{level}.tdb", schema)
     if created:
         # write image and metadata to TileDB array
         with open_bioimg(uri, "w") as out_array:
             out_array.meta.update(reader.level_metadata(level), level=level)
+            min_max_axes = tuple(
+                idx for idx in range(len(source_axes.dims)) if idx != channel_index
+            )
+
             if chunked or max_workers:
                 inv_axes_mapper = axes_mapper.inverse
+                min_max_lock = Lock()
 
-                def tile_to_tiledb(level_tile: Tuple[slice, ...]) -> None:
+                def tile_to_tiledb(level_tile: Tuple[slice, ...], lock: Lock) -> None:
                     source_tile = inv_axes_mapper.map_tile(level_tile)
                     image = reader.level_image(level, source_tile)
                     out_array[level_tile] = axes_mapper.map_array(image)
 
+                    minimum = np.amin(image, axis=min_max_axes)
+                    maximum = np.amax(image, axis=min_max_axes)
+
+                    with lock:
+                        for channel in range(channel_count):
+                            channel_min_max[channel][0] = min(
+                                channel_min_max[channel][0], minimum.item(channel)
+                            )
+                            channel_min_max[channel][1] = max(
+                                channel_min_max[channel][1], maximum.item(channel)
+                            )
+
```

Comment:
This segment here is repeated in `L:439-L:450`. Could you this avoid code repetition?